contracts/AssetPool.sol

```diff
@@ -17,16 +18,19 @@ import "./UnderwriterToken.sol";
 ///         in KEEP in return for covKEEP underwriter tokens. Underwriter tokens
 ///         represent an ownership share in the underlying collateral of the
 ///         Asset Pool.
-contract AssetPool is Ownable {
+contract AssetPool is Ownable, IAssetPool {
     using SafeERC20 for IERC20;
     using SafeERC20 for UnderwriterToken;
     using SafeMath for uint256;
 
     IERC20 public collateralToken;
     UnderwriterToken public underwriterToken;
+    UnderwriterToken public newUnderwriterToken;
```

Comment:
We could have a `depositFor(uint256 amount, address covRecipient)` function. This would simplify the design - we wouldn't have to compute the balance of new COV when updating and we wouldn't need this reference.

Reply:
I think that was the initial approach, but based on [this](https://www.flowdock.com/app/cardforcoin/tech/threads/kLsWXXDV8tcJ2uXtKBQqxqDAUPA) conversation, you mentioned we won't need to pass underwriter and that's why we need `UnderwriterToken` here to make it happen.

Okay, I'll change it to the initial approach.